collector/odigosotelcol/go.mod

```diff
@@ -2,7 +2,9 @@
 
 module odigos.io/opentelemetry-collector/cmd/odigosotelcol
 
-go 1.23.0
+go 1.23.3
+
+toolchain go1.23.4
```

Comment:
@RonFed fyi this happened switching to the vitess library:

```
$ make go-mod-tidy
go: finding module for package vitess.io/vitess/go/vt/sqlparser
go: downloading vitess.io/vitess v0.21.1
go: toolchain upgrade needed to resolve vitess.io/vitess/go/vt/sqlparser
go: vitess.io/vitess@v0.21.1 requires go >= 1.23.3; switching to go1.23.4
go: downloading go1.23.4 (darwin/arm64)
go: finding module for package vitess.io/vitess/go/vt/sqlparser
go: found vitess.io/vitess/go/vt/sqlparser in vitess.io/vitess v0.21.1
```

Reply:
I think we can remove the `toolchain` line, run `go mod tidy `again and it won't require the `toolchain` part but I'm not sure.